src/hooks/useEventEmitter.ts

```diff
@@ -33,10 +32,25 @@ export function useTypedEventEmitter<
 }
 
 // Hook to wrap event emitter on and removeListener in hook lifecycle
+export function useEventEmitter(
+    emitter: EventTarget | undefined,
+    eventName: string,
+    handler: Handler,
+): void;
 export function useEventEmitter(
     emitter: EventEmitter | undefined,
     eventName: string | symbol,
     handler: Handler,
+): void;
+export function useEventEmitter(
+    emitter: EventEmitter | EventTarget | undefined,
+    eventName: string | symbol,
+    handler: Handler,
+): void;
```

Comment:
This one is superfluous, it is identical to the implementation signature, can we also use `Optional<>` instead of `| undefined` please

Reply:
Without this, the call from `useEventEmitterState()` will throw a type error. See https://matrix.to/#/!bEWtlqtDwCLFIAKAcv:matrix.org/$gARGyprqMwHqkQXJFrNgRUjB5kX9ScLle7OD8ih-sUI?via=matrix.org&via=element.io&via=chat.weho.st

Reply:
Okay but it is also invalid, it says that you can give an EventTarget + symbol and it'll be fine when in reality it won't be accepted.

![image](https://user-images.githubusercontent.com/2403652/179772408-5b80f4d3-2c36-4164-a3b6-db5769ba3196.png)

Should error. Is there any benefit from loading this into the same hook into of a `useEventTarget`, `useEventTargetState` hooks?